src/mainviews/job-detail.tsx

```diff
@@ -120,201 +96,235 @@ export function JobDetail(props: IJobDetailProps): JSX.Element {
+              | React.MouseEvent<HTMLSpanElement, MouseEvent>
+          ): void => props.setView('ListJobs')}
+        >
+          {trans.__('Notebook Jobs')}
+        </Link>
+        <Typography color="text.primary">{props.model.jobName}</Typography>
+      </Breadcrumbs>
+    </div>
+  );
+
+  const ButtonBar = () => (
+    <Stack direction="row" gap={2} justifyContent="flex-end" flexWrap={'wrap'}>
+      {props.model.status === 'IN_PROGRESS' && (
+        <Button variant="outlined" onClick={handleStopJob}>
+          {trans.__('Stop Job')}
+        </Button>
+      )}
+      <Button variant="outlined" onClick={handleRerunJob}>
+        {trans.__('Rerun Job')}
+      </Button>
+      <Button variant="contained" color="error" onClick={handleDeleteJob}>
+        {trans.__('Delete Job')}
+      </Button>
+    </Stack>
+  );
+
+  const coreOptionsFields: TextFieldProps[][] = [
+    [
+      { defaultValue: props.model.jobName, label: trans.__('Job name') },
+      { defaultValue: props.model.jobId, label: trans.__('Job ID') }
+    ],
+    [
+      {
+        defaultValue: props.model.inputFile,
+        label: trans.__('Input file')
+      },
+      {
+        defaultValue: props.model.outputPath,
+        label: trans.__('Output path')
+      }
+    ],
+    [
+      {
+        defaultValue: props.model.environment,
+        label: trans.__('Environment')
+      },
+      { defaultValue: props.model.status ?? '', label: trans.__('Status') }
+    ],
+    [
+      {
+        defaultValue: timestampLocalize(props.model.createTime ?? ''),
+        label: trans.__('Created at')
+      },
+      {
+        defaultValue: timestampLocalize(props.model.updateTime ?? ''),
+        label: trans.__('Updated at')
+      }
+    ],
+    [
+      {
+        defaultValue: timestampLocalize(props.model.startTime ?? ''),
+        label: trans.__('Start time')
+      },
+      {
+        defaultValue: timestampLocalize(props.model.endTime ?? ''),
+        label: trans.__('End time')
+      }
+    ]
+  ];
+
+  function OutputFile(props: {
+    outputType: string;
+    app: JupyterFrontEnd;
+    outputPath: string;
+  }) {
+    // Get all output files.
+    const outputName = props.outputPath.replace(/ipynb$/, props.outputType);
     return (
-      <TextField
-        {...props}
-        label={props.label}
-        defaultValue={props.defaultValue}
-        variant="outlined"
-        disabled
-      />
+      <Link
+        key={props.outputType}
+        href={`/lab/tree/${outputName}`}
+        title={trans.__('Open "%1"', outputName)}
+        onClick={(
+          e:
+            | React.MouseEvent<HTMLSpanElement, MouseEvent>
+            | React.MouseEvent<HTMLAnchorElement, MouseEvent>
+        ) => {
+          e.preventDefault();
+          props.app.commands.execute('docmanager:open', {
+            path: outputName
+          });
+        }}
+        style={{ paddingRight: '1em' }}
+      >
+        {props.outputType}
+      </Link>
     );
```

Comment:
This looks like a duplication of https://github.com/jupyter-server/jupyter-scheduler/blob/9623dceb73889fd44689208d7bcb32888521e854/src/components/job-row.tsx#L133-L164. Can you re-use this?

Reply:
It's essentially the same function adapted to the Job Detail view needs / types used in the view. Original function in `job-row` accepts `Scheduler.IDescribeJob`, type that API call returns while adapted function (in job-detail) accepts `IJobDetailModel`, same type it's getting passed to and maps all its fields from. I do call API from inside the function so as an option I could save API return object in the state but it would somewhat duplicate the state and go against the principle that this component works with "front-end types"

Reply:
Job Detail doesn't have anything additional that list doesn't have.  Which API call are you referring to?

Reply:
To the call currently at line 56. Created [issue 56](https://github.com/jupyter-server/jupyter-scheduler/issues/75) to track it